src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -337,44 +339,67 @@ internal partial class DownstreamApi : IDownstreamApi
 
             return downstreamApiResult;
         }
-
-        internal async Task UpdateRequestAsync(
+
+        internal /* internal for test */ async Task UpdateRequestAsync(
             HttpRequestMessage httpRequestMessage,
             HttpContent? content,
             DownstreamApiOptions effectiveOptions,
             bool appToken,
             ClaimsPrincipal? user,
             CancellationToken cancellationToken)
         {
+            AddCallerSDKTelemetry(effectiveOptions);
+
             if (content != null)
             {
                 httpRequestMessage.Content = content;
             }
 
             effectiveOptions.RequestAppToken = appToken;
 
-            // Obtention of the authorization header (except when calling an anonymous endpoint
-            // which is done by not specifying any scopes
-            if (effectiveOptions.Scopes != null && effectiveOptions.Scopes.Any())
-            {
+            // Obtention of the authorization header (except when calling an anonymous endpoint
+            // which is done by not specifying any scopes
+            if (effectiveOptions.Scopes != null && effectiveOptions.Scopes.Any())
+            {
                 string authorizationHeader = await _authorizationHeaderProvider.CreateAuthorizationHeaderAsync(
                        effectiveOptions.Scopes,
                        effectiveOptions,
                        user,
-                       cancellationToken).ConfigureAwait(false);
-                
+                       cancellationToken).ConfigureAwait(false);
+
                 httpRequestMessage.Headers.Add(Authorization, authorizationHeader);
-            }
-            else
-            {
-                Logger.UnauthenticatedApiCall(_logger, null);
-            }
-            if (!string.IsNullOrEmpty(effectiveOptions.AcceptHeader))
+            }
+            else
+            {
+                Logger.UnauthenticatedApiCall(_logger, null);
+            }
+            if (!string.IsNullOrEmpty(effectiveOptions.AcceptHeader))
             {
                 httpRequestMessage.Headers.Accept.ParseAdd(effectiveOptions.AcceptHeader);
-            }
-            // Opportunity to change the request message
+            }
+            // Opportunity to change the request message
             effectiveOptions.CustomizeHttpRequestMessage?.Invoke(httpRequestMessage);
-        }
-    }
-}
+        }
+
+        internal /* for test */ static Dictionary<string, string> CallerSDKDetails { get; } = new()
+          {
+              { "caller-sdk-id", "1" },  // 1 = Downstream API SDK ID
+              { "caller-sdk-ver", IdHelper.GetIdWebVersion() }
+          };
+
+        private static void AddCallerSDKTelemetry(DownstreamApiOptions effectiveOptions)
+        {
+            if (effectiveOptions.AcquireTokenOptions.ExtraQueryParameters == null)

```

Comment:
@bgavrilMS 
I thought we had agreed to use a new collection `Dictionary<string, object> ExtraParameters`  not Extra query parameters (unless this is really what MSAL sends as is, but I don't think it's the case?

Reply:
We haven't agreed on a Dictionary<string, object> in MSAL, you proposed a strongly typed ext mechanism - https://github.com/AzureAD/microsoft-identity-web/issues/2975

ExtraQP is the simplest extensibility mechanism we have. The work is already done. What are the concerns?